Bowling/Bowling/Score.cs

```diff
@@ -0,0 +1,167 @@
+                    }
+                    else if (frameMarks.Length == 2)
+                    {
+                        bowlingMarksByFrame.Add(frameMarks);
+                        frameMarks = String.Empty;
+                        frameCount++;
+                    }
+                }
+            }
+
+            bowlingMarksByFrame.Add(frameMarks);
+            return bowlingMarksByFrame;
+        }
+
+
+        private List<Frame> UpdateFrameScoresForStrikesAndSpares(List<Frame> framesCollection)
+        {
+            var frameNumber = 1;
+            
+            foreach(Frame frame in framesCollection)
+            {
+                var frameScore = frame.frameScore;
+                
+                if(frame.frameOutcome.Equals(RollTypes.strike))
+                {
+                    frameScore = UpdateFrameScoreForAStrike(framesCollection, frameNumber);
+                }
+                else if(frame.frameOutcome.Equals(RollTypes.spare))
+                {
+                    frameScore = UpdateFrameScoreForASpare(framesCollection, frameNumber);
+                }
+                
+                frame.frameScore = frameScore;
+                frameNumber++;
+            }
+
+            return framesCollection;
+        }
+
+        private int UpdateFrameScoreForAStrike(List<Frame> framesCollection, int frameNumber)
+        {
+            if (frameNumber != _frameNumberTotal)
+            {
+                Frame currentFrame = framesCollection[frameNumber - 1];
+                Frame oneFrameAfterCurrentFrame = framesCollection[frameNumber];
+
+                if (oneFrameAfterCurrentFrame.frameOutcome == RollTypes.strike)
+                {
+                    Frame twoFramesAfterCurrentFrame = framesCollection[frameNumber+1];
+                    return currentFrame.frameScore + oneFrameAfterCurrentFrame.firstRoll + twoFramesAfterCurrentFrame.firstRoll;
+                }
+                else
+                {
+                    return currentFrame.frameScore + oneFrameAfterCurrentFrame.firstRoll + oneFrameAfterCurrentFrame.secondRoll;
+                }
+            }
+            else
+            {
+                return framesCollection[frameNumber - 1].frameScore;
+            }
+        }
+
+        private int UpdateFrameScoreForASpare(List<Frame> framesCollection, int frameNumber)
+        {
+            if (frameNumber != _frameNumberTotal)
+            {
+                Frame currentFrame = framesCollection[frameNumber - 1];
+                Frame oneFrameAfterCurrentFrame = framesCollection[frameNumber];
+
+                return currentFrame.frameScore = currentFrame.frameScore + oneFrameAfterCurrentFrame.firstRoll;
+            }
+            else
+            {
+                return framesCollection[frameNumber - 1].frameScore;
+            }
+        }
+
+        private int CalculateFramesGrandTotal(List<Frame> framesCollection)
+        {
+           var grandTotal = 0;
```

Comment:
LINQ is your friend here (I think you have access to this by default, if not, ignore this comment).

```
private int CalculateFramesGrandTotal(IEnumerable<Frame> framesCollection)
{
    return framesCollection.Sum(frame => frame.frameScore);
}
```
...or...if you end up removing `frameScore` from that model:
```
private int CalculateFramesGrandTotal(IEnumerable<Frame> framesCollection)
{
    return framesCollection.Sum(frame => { frame.firstRoll + frame.secondRoll + frame.bonusRoll });
}
```

Reply:
Updated this with LINQ sum.  Much nicer!